.github/workflows/test.yml

```
@@ -18,15 +18,19 @@ jobs:
 
     strategy:
       matrix:
-        os: [windows-latest, macos-latest, ubuntu-latest]
         include:
           # version number must be string, otherwise 3.10 becomes 3.1
           - os: windows-latest
             python-version: "3.11"
+            installs: "numpy>=2.0.0rc1"
           - os: macos-latest
             python-version: "3.8"
+            installs: "'numpy==1.21.0'"
           - os: ubuntu-latest
             python-version: "pypy-3.8"
+          - os: ubuntu-latest
+            python-version: "3.12"
+            installs: "'numpy>=2.0.0rc1' 'scipy>=1.13.0'"
```

```suggestion
            installs: "'numpy>=2.0.0rc1' scipy matplotlib"
```

Matplotlib 3.8.4 just came out, which supports numpy 2. Also, there's no need to put a `>=` here unless there are only dev releases, so I removed the `>=1.13` on scipy.